qiskit/qpy/binary_io/circuits.py

```diff
@@ -294,6 +297,14 @@ def _read_instruction(
     # Load Parameters
     for _param in range(instruction.num_parameters):
         type_key, data_bytes = common.read_generic_typed_data(file_obj)
+        # If the gate is in the list of exceptions (subclasses of BlueprintCircuit),
+        # and the parameter type is a parameter vector, initialize all elements of
+        # the vector. This is to avoid unnecessary UserWarnings about elements in
+        # the non-user-actionable parameter vector not being initialized.
+        initialize_full_vec = any(
+            exception in gate_name
+            for exception in ["ZZFeatureMap", "ZFeatureMap", "PauliFeatureMap"]
```

Comment:
This can also be triggered outside of these classes, e.g. running Imamichi-san's example with the following, manually built circuit, also gives the `UserWarning`:
```python
def manual():
    x = ParameterVector("γ", 2)
    y = ParameterVector("γ", 2)
    # new construction
    ansatz = QuantumCircuit(3)
    ansatz.rzz(2 * y[0], 0, 1)

    ansatz.assign_parameters({ansatz.parameters[0]: x[0]}, inplace=True)
    qc = QuantumCircuit(3)
    qc.append(ansatz, range(3))

    return qc
```

It would be nice if we don't have to rely on hardcoding, too. Would it make sense to just always do the full initialization?